File: yts_project_vueai/src/api/modules/adminOperationLog.ts
import request from '../request'

export interface AdminOperationLogItem {
  id: number
  operatorId?: number
  operatorName?: string
  operatedAt: string
  operationType: string
  module: string
  targetType?: string
  targetId?: string
  targetLabel?: string
  changeSummary?: string
  changeDetail?: string
  ip?: string
  userAgent?: string
}

export interface OperationLogListParams {
  operatorId?: number
  module?: string
  operationType?: string
  from?: string
  to?: string
  pageNo?: number
  pageSize?: number
}

export interface OperationLogListResult {
  list: AdminOperationLogItem[]
  total: number
  pageNo: number
  pageSize: number
  totalPages: number
}

export const adminOperationLogApi = {
  list: async (params?: OperationLogListParams): Promise<OperationLogListResult> => {
    const response = await request({
      url: '/admin-operation-log/list',
      method: 'GET',
      params
    })
    const data = response as OperationLogListResult
    return {
      list: Array.isArray(data?.list) ? data.list : [],
      total: data?.total ?? 0,
      pageNo: data?.pageNo ?? 1,
      pageSize: data?.pageSize ?? 20,
      totalPages: data?.totalPages ?? 0
    }
  }
}

File: yts_project/src/main/java/com/it/yts_project/model/AdminOperationLog.java
package com.it.yts_project.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

import java.time.LocalDateTime;

/**
 * 后台操作日志：记录谁在何时对哪块数据做了何种操作，供管理员审计
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminOperationLog {

    private Long id;
    private Integer operatorId;
    private String operatorName;
    private LocalDateTime operatedAt;
    private String operationType;
    private String module;
    private String targetType;
    private String targetId;
    private String targetLabel;
    private String changeSummary;
    private String changeDetail;
    private String ip;
    private String userAgent;
}
